app/models/spree/variant.rb

```diff
@@ -114,7 +114,7 @@ class Variant < ActiveRecord::Base
       joins("
         LEFT OUTER JOIN (SELECT *
                            FROM inventory_items
-                           WHERE enterprise_id = #{sanitize enterprise.andand.id})
+                           WHERE enterprise_id = #{enterprise.andand.id})
```

Comment:
This should be fine but makes me slightly nervous. Can we use `enterprise_id` here and define that above?
```rb
enterprise_id = enterprise&.id.to_i
return none if enterprise_id < 1
```

Reply:
We already have this at the start of the scope here:
```ruby
return where("1=0") if enterprise.blank?
```
Do we need more than that..?

Reply:
My suggestion is to call `to_i` so that we are definitely safe from SQL injection. The "return none..." part would just replace the current "return where...` line.

Just checking that `enterprise` is present doesn't prove that it's an `Enterprise` object and that it returns a number when calling `id`.

Reply:
I see this scope is only used in 3 places. We could perhaps ensure it's indeed an enterprise object there.

Reply:
Added :+1: